pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -283,9 +293,57 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		Message: machineCreationSucceedMessage,
 	})
 
+	s.setMachineCloudProviderSpecifics(vm)
+
 	return nil
 }
 
+func getVMState(vm compute.VirtualMachine) v1beta1.VMState {
+	if vm.ProvisioningState == nil {
+		return ""
+	}
+
+	if *vm.ProvisioningState != "Succeeded" {
+		return v1beta1.VMState(*vm.ProvisioningState)
+	}
+
+	if vm.InstanceView != nil && vm.InstanceView.Statuses != nil {
+		for _, status := range *vm.InstanceView.Statuses {
+			if status.DisplayStatus == nil || *status.DisplayStatus == "Provisioning succeeded" {
+				continue
+			}
+			// "Running" string is sufficient
+			if *status.DisplayStatus == "VM running" {
```

Comment:
what's the benefit of introducing this new state?
This is breaking `exists` since is not accounted for https://github.com/openshift/cluster-api-provider-azure/pull/59/files#diff-7c34c0cb525c96c7b732376b0c8994e1L298

Reply:
> what's the benefit of introducing this new state?

Since when an instance is running, you want to know it's running? 

> This is breaking exists() since is not accounted for https://github.com/openshift/cluster-api-provider-azure/pull/59/files#diff-7c34c0cb525c96c7b732376b0c8994e1L298

Good catch.

Reply:
Can we please rely on `status.code` instead of `DisplayStatus` e.g `PowerState/running` and put it in a constant?
Also can we please unit test this func.
See https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/virtual-machines-common-states-lifecycle.md for reference. Also `az vm get-instance-view --resource-group foo --name bar` might come in handy

Reply:
Done